### PR TITLE
Updating System.Text.Json package description

### LINKED ISSUE #38624: The System.Text.Json package description should be updated
The `System.Text.Json` package description is now a bit out-dated and we should update it to reflect the contents more accurately.

https://github.com/dotnet/corefx/blob/ef0578fd6abeba406ddffbb1591c1a618bca9dae/pkg/descriptions.json#L1997-L2013

Current view on nuget.org: https://www.nuget.org/packages/System.Text.Json/4.6.0-preview6.19303.8
image

This includes adding some of the newer components like `JsonSerializer`, `JsonElement`, and maybe even the converter types once they show up.

We should also remove the (now unnecessary) `Microsoft.Bcl.Json.Sources` content.

I am filing an issue primarily to track this gets done before release and so that I don't forget it.

cc @steveharter, @joperezr

### PR DESCRIPTION
**Pull request by joperezr**
fixes #38624

cc: @ahsonkhan

### CODE REVIEW

File: pkg/descriptions.json [hunk 1]
```diff
@@ -1997,18 +1997,11 @@
...
-  },
-  {
-    "Name": "Microsoft.Bcl.Json.Sources",
-    "Description": "Provides the source code implementation of System.Text.Json as internal types. This package is only required for projects targeting .NET Framework 4.6.1+ or .NET Core 2.x; and is not for projects that can target .NET Core 3.0 and higher. For guidance on consuming the source pacakge, visit: https://aka.ms/json-source-package-guide.",
-    "CommonTypes": [
-      "System.Text.Json.Utf8JsonReader",
-      "System.Text.Json.Utf8JsonWriter",
-      "System.Text.Json.JsonDocument"
+      "System.Text.Json.JsonElement",
+      "System.Text.Json.Serialization.JsonSerializer"
```
Comment (stephentoub) [inline pkg/descriptions.json]: It's no longer in the Serialization namespace, is it?

File: pkg/descriptions.json [hunk 2]
```diff
@@ -1997,18 +1997,11 @@
     "Name": "System.Text.Json",
     "Description": "Provides a fast, non-cached, forward-only way to read UTF-8 encoded JavaScript Object Notation (JSON) text.",
     "CommonTypes": [
+      "System.Text.Json.JsonDocument",
```
Comment (stephentoub) [inline pkg/descriptions.json]: Should these be in alphabetical order?

File: pkg/descriptions.json (hunk 1 above)
Comment (joperezr) [reply pkg/descriptions.json]: mm it does seem like it is:

https://github.com/dotnet/corefx/blob/1c1b78f7ec1d04aeac059d3d4e4c975d3a3ad23e/src/System.Text.Json/ref/System.Text.Json.cs#L361-L405

File: pkg/descriptions.json (hunk 2 above)
Comment (joperezr) [reply pkg/descriptions.json]: mm good question, I don't think we follow patterns here as I have found both cases. Do you have a specific preference? @terrajobst @ahsonkhan

File: pkg/descriptions.json (hunk 1 above)
Comment (stephentoub) [reply pkg/descriptions.json]: Look at master.

Comment (ahsonkhan) [reply pkg/descriptions.json]: @stephentoub's correct. The `JsonSerializer` is in the S.T.Json namespace now.

File: pkg/descriptions.json (hunk 2 above)
Comment (ahsonkhan) [reply pkg/descriptions.json]: I am fine with ordering it as what would be most commonly used.

1) JsonSerializer
2) JsonDocument
3) JsonElement
4) Utf8JsonWriter
5) Utf8JsonReader

File: pkg/descriptions.json [hunk 3]
```diff
@@ -1997,18 +1997,11 @@
     "Name": "System.Text.Json",
     "Description": "Provides a fast, non-cached, forward-only way to read UTF-8 encoded JavaScript Object Notation (JSON) text.",
```
Comment (ahsonkhan) [inline pkg/descriptions.json]: We should update the description too to make it broader, rather than specific to the reader.

Comment (joperezr) [reply pkg/descriptions.json]: Do you have a suggestion? I'm not very good at writing descriptions 😄 Perhaps something like:
```
Provides types that help read/write/serialize/deserialize UTF-8 encoded JavaScript Object Notation (JSON) text.
```

Comment (ahsonkhan) [reply pkg/descriptions.json]: I took a look at some of the XML namespaces and their description.
> "Provides classes for serializing objects to the Extensible Markup Language (XML) and deserializing XML data to objects."

https://github.com/dotnet/corefx/blob/c53300f0696c4817bc06a8ef8711a0bb46f362ec/pkg/descriptions.json#L2199

> "Provides an older in-memory Extensible Markup Language (XML) programming interface that enables you to modify XML documents. Developers should prefer the classes in the System.Xml.XDocument package."

> "Provides classes for serializing objects to the Extensible Markup Language (XML) and deserializing XML data to objects."

**Maybe something like this (if it is not too verbose):**
Provides types to serialize objects to JavaScript Object Notation (JSON) text and deserialize JSON data to objects. Also provides types to read and write JSON encoded as UTF-8, and to create an in-memory document object model (DOM) that is read-only, to get random access of JSON elements within a structured view of the JSON.

**Pull request merged**